devices/IFrameTransformToIWear/src/IFrameTransformToIWear.cpp

```diff
@@ -0,0 +1,489 @@
+/*
+ * Copyright (C) 2021 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include "IFrameTransformToIWear.h"
+
+#include <yarp/dev/IFrameTransform.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/sig/Vector.h>
+
+#include <map>
+#include <mutex>
+#include <string>
+#include <vector>
+
+const std::string DeviceName = "IFrameTransformToIWear";
+const std::string LogPrefix = DeviceName + " :";
+
+using namespace wearable;
+using namespace wearable::sensor;
+using namespace wearable::devices;
+
+// Class that provides each frame information, and utilites
+// to map its data to containers compatible with IWear
+class IFrameTransformHandler
+{
+public:
+    IFrameTransformHandler(const std::string& targetFrame,
+                           const std::string& rootFrame,
+                           yarp::dev::IFrameTransform* frameTransformInterface)
+        : m_targetFrame(targetFrame)
+        , m_rootFrame(rootFrame)
+        , m_frameTransformInterface(frameTransformInterface)
+        , m_matrixBuffer(yarp::sig::Matrix(4, 4)){};
```

Comment:
does this class has a direct constructor which is not involve a copy?

Reply:
I mean for `m_matrixBuffer`

Reply:
Updated in https://github.com/robotology/wearables/pull/126/commits/2ce93cfdabaa908d061e5f8df589c51abf2c1c0f, is that what you meant @diegoferigo right?

Reply:
Yep!